### PR TITLE
Update test suite to use new reactphp/async package instead of clue/reactphp-block

### PR DESCRIPTION
**Pull request by WyriHaximus**
See https://github.com/reactphp/async
Refs https://github.com/reactphp/socket/pull/296

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -21,7 +21,7 @@
         "wyrihaximus/react-child-process-promise-closure": "^1.0"
     },
     "require-dev": {
-        "clue/block-react": "^1.4",
+        "react/async": "^4 || ^3",
         "phpunit/phpunit": "^9.5"
```
Comment (clue) [inline composer.json]: nit: Should be sorted by name?

**Review by clue — COMMENTED** [review]
:+1:

File: composer.json (hunk 1 above)
Comment (WyriHaximus) [reply composer.json]: Done :+1:

**Review by clue — APPROVED** [review]
@WyriHaximus Thanks for looking into this, changes LGTM! :shipit:

**Pull request merged**